pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,579 @@
+        :math:`j\in\left\{1,\ldots,n\right\}`, and :math:`r_{i}` for all
+        :math:`i\in\left\{1,\ldots,2n\right\}`.
+
+        .. math::
+
+            \begin{bmatrix}
+            x_{11} & \cdots & x_{1n} &        & z_{11} & \cdots & z_{1n} & &r_{1}\\
+            \vdots & \ddots & \vdots & & \vdots & \ddots & \vdots & &\vdots\\
+            x_{n1} & \cdots & x_{nn} &        & z_{n1} & \cdots & z_{nn} & &r_{n}\\
+            & & & & & & & & \\
+            x_{\left(  n+1\right)  1} & \cdots & x_{\left(  n+1\right)  n} & &
+            z_{\left(  n+1\right)  1} & \cdots & z_{\left(  n+1\right)  n} & & r_{n+1}\\
+            \vdots & \ddots & \vdots  & & \vdots & \ddots & \vdots & & \vdots\\
+            x_{\left(  2n\right)  1}  & \cdots & x_{\left(  2n\right)  n} & &
+            z_{\left(  2n\right)  1}  & \cdots & z_{\left(  2n\right)  n} & & r_{2n}
+            \end{bmatrix}
+
+        Rows :math:`1` to :math:`n` of the tableau represent the destabilizer generators
+        :math:`R_{1},\ldots,R_{n}`, and rows :math:`n+1` to :math:`2n` represent the stabilizer
+        generators :math:`R_{n+1},\ldots,R_{2n}`. If :math:`R_{i}=\pm P_{1}\ldots P_{n}`,
+        then bits :math:`x_{ij},z_{ij}` determine the j\ :math:`^{th}` Pauli matrix
+        :math:`P_{j}:\ 00` means ``I``, :math:`01` means ``X``, :math:`11` means ``Y``,
+        and :math:`10` means ``Z``. Finally, :math:`r_{i}` is one if :math:`R_{i}`
+        has negative phase and zero if :math:`r_{i}` has a positive phase.
+
+
+    .. details::
+        :title: Tracking
+        :href: clifford-tracking
+
+        ``DefaultClifford`` tracks:
+
+        * ``executions``: the number of unique circuits that would be required on quantum hardware
+        * ``shots``: the number of shots
+        * ``resources``: the :class:`~.resource.Resources` for the executed circuit.
+        * ``simulations``: the number of simulations performed. One simulation can cover multiple QPU executions,
+          such as for non-commuting measurements and batched parameters.
+        * ``batches``: The number of times :meth:`~.execute` is called.
+        * ``results``: The results of each call of :meth:`~.execute`.
+
+    .. details::
+        :title: Accelerate calculations with multiprocessing
+        :href: clifford-multiprocessing
+
+        See the details in :class:`~pennylane.devices.DefaultQubit`'s "Accelerate calculations with multiprocessing"
+        section. Additional information regarding multiprocessing can be found in the
+        `multiprocessing doc <https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods>`_.
+    """
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.clifford"
+
+    # pylint:disable = too-many-arguments
+    def __init__(
+        self,
+        wires=None,
+        shots=None,
+        check_clifford=True,
+        tableau=True,
```

Comment:
Maybe it's already been discussed. But do we prefer the default to be `tableau=True` or `tableau=False`?
If I didn't read the documentation carefully, I would be surprised when I `return qml.state()` and get something that has `np.shape` = `(40,41)`

Reply:
I think the general consensus was to go with `tableau=True` for the sake of efficiency (and we document it properly). If users try to compute the state of a circuit with many qubits, they will be better off seeing a square matrix than a frozen system.